Currency management in column amount

programmer4web opened the pull request:
1. Rates object, getComparableValue updated ( sorting and filtering are now based on amount value ).  added tests with rates.
2. Configurable 'currency' property used in filters menu, added test.
3. Auto-detect 'currency' property if not configured.
4. If 'autodetect' property is true then 'currency' with be the currency with the highest occurrence ( implemented in function _valuesChanged ). added tests.
5. formatters optimization: when data changes get only formatters that are not already created from the old set of values.
6. Seems there is still an issue, in the input fields, min and max values are not converted to the detected currency.

fixes https://github.com/Neovici/cosmoz-omnitable/issues/24

programmer4web commented:
Currency management is ready for review. @nomego

plequang commented on cosmoz-omnitable-column-amount.html:
```diff
@@ -49,9 +48,23 @@ <h3 style="margin: 0;">[[ title ]]</h3>
+					type: Boolean,
+					value: false
+				},
+				rates: {
```
When a code comment  here to explain what this object should contain

programmer4web replied:
Rates object is for currency exchange rates ( In this example Units to EUR is used.)
example is placed in demo rates='{"EUR": 1, "USD":0.8169982616, "AUD":0.6529827192, "SEK": 0.1019271438}'

Data from: 
http://www.xe.com/currencyconverter/convert/?Amount=1&From=USD&To=EUR
http://www.xe.com/currencyconverter/convert/?Amount=1&From=AUD&To=EUR
http://www.xe.com/currencyconverter/convert/?Amount=1&From=SEK&To=EUR

I will add more details in code also.

plequang replied:
Yes I was thinking of a code comment that would appear in the api http://127.0.0.1:8081/components/cosmoz-omnitable/#/elements/cosmoz-omnitable-column-amount

plequang commented:
@programmer4web  Did you see this comment https://github.com/Neovici/cosmoz-omnitable/pull/143#pullrequestreview-101170315 ?

programmer4web commented:
@plequang yes I will add tests that fails with this case and then an fix for toAmount(value, limit, limitFunc) (I was working on code coverage pull requests)

programmer4web commented:
@plequang I have added tests for toAmount function with suggested case and updated the function to compare amounts with rates. This is ready for review.

programmer4web commented:
@plequang Can you review this again please about min, max of input conversion to currency. The actual limiting of input to min, max in in another PR https://github.com/Neovici/cosmoz-omnitable/pull/131

programmer4web commented:
@plequang I have rebased this. Can you review it again please.
